components/layout_2020/taffy/layout.rs

```diff
@@ -0,0 +1,620 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::marker::PhantomData;
+
+use app_units::Au;
+use atomic_refcell::{AtomicRef, AtomicRefCell, AtomicRefMut};
+use style::properties::ComputedValues;
+use style::values::generics::length::{GenericLengthPercentageOrAuto, LengthPercentageOrAuto};
+use style::values::specified::align::AlignFlags;
+use style::values::specified::box_::DisplayInside;
+use style::Zero;
+use taffy::style_helpers::{TaffyMaxContent, TaffyMinContent};
+use taffy::MaybeMath;
+
+use super::{TaffyContainer, TaffyItemBox, TaffyItemBoxInner, TaffyStyloStyle};
+use crate::cell::ArcRefCell;
+use crate::context::LayoutContext;
+use crate::formatting_contexts::{Baselines, IndependentFormattingContext, IndependentLayout};
+use crate::fragment_tree::{BoxFragment, CollapsedBlockMargins, Fragment};
+use crate::geom::{
+    AuOrAuto, LogicalSides, LogicalVec2, PhysicalPoint, PhysicalRect, PhysicalSides, PhysicalSize,
+    Size,
+};
+use crate::positioned::{AbsolutelyPositionedBox, PositioningContext, PositioningContextLength};
+use crate::sizing::{ContentSizes, InlineContentSizesResult};
+use crate::style_ext::ComputedValuesExt;
+use crate::{ContainingBlock, IndefiniteContainingBlock};
+
+const DUMMY_NODE_ID: taffy::NodeId = taffy::NodeId::new(u64::MAX);
+
+fn resolve_content_size(constraint: taffy::AvailableSpace, content_sizes: ContentSizes) -> f32 {
+    match constraint {
+        taffy::AvailableSpace::Definite(limit) => {
```

Comment:
I would import `taffy::AvailableSpace` here unless it conflicts with some other definition.